test/e2e/operator_test.go

```diff
@@ -0,0 +1,294 @@
+
+				// E.g. IMAGE_FORMAT=registry.build03.ci.openshift.org/ci-op-52fj47p4/stable:${component}
+				registry := strings.Split(os.Getenv("IMAGE_FORMAT"), "/")[0]
+
+				required.Spec.Template.Spec.Containers[0].Image = registry + "/" + os.Getenv("NAMESPACE") + "/pipeline:run-once-duration-override-operator"
+				_, _, err := resourceapply.ApplyDeployment(
+					ctx,
+					kubeClient.AppsV1(),
+					eventRecorder,
+					required,
+					1000, // any random high number
+				)
+				return err
+			},
+		},
+		{
+			path: "assets/08_crd.yaml",
+			readerAndApply: func(objBytes []byte) error {
+				_, _, err := resourceapply.ApplyCustomResourceDefinitionV1(ctx, apiExtClient.ApiextensionsV1(), eventRecorder, resourceread.ReadCustomResourceDefinitionV1OrDie(objBytes))
+				return err
+			},
+		},
+		{
+			path: "assets/09_cr.yaml",
+			readerAndApply: func(objBytes []byte) error {
+				requiredObj, err := runtime.Decode(runoncedurationoverridescheme.Codecs.UniversalDecoder(runoncedurationoverridev1.SchemeGroupVersion), objBytes)
+				if err != nil {
+					klog.Errorf("Unable to decode assets/09_cr.yaml: %v", err)
+					return err
+				}
+				requiredSS := requiredObj.(*runoncedurationoverridev1.RunOnceDurationOverride)
+
+				_, err = runOnceDurationOverrideClient.RunOnceDurationOverrideV1().RunOnceDurationOverrides().Create(ctx, requiredSS, metav1.CreateOptions{})
+				return err
+			},
+		},
+	}
+
+	// create required resources, e.g. namespace, crd, roles
+	if err := wait.PollImmediate(1*time.Second, 10*time.Second, func() (bool, error) {
+		for _, asset := range assets {
+			klog.Infof("Creating %v", asset.path)
+			if err := asset.readerAndApply(bindata.MustAsset(asset.path)); err != nil {
+				klog.Errorf("Unable to create %v: %v", asset.path, err)
+				return false, nil
+			}
+		}
+
+		return true, nil
+	}); err != nil {
+		klog.Errorf("Unable to create SSO resources: %v", err)
+		os.Exit(1)
+	}
+
+	var runOnceDurationoverridePod *corev1.Pod
+	// Wait until the run once duration override pod is running
+	if err := wait.PollImmediate(5*time.Second, 1*time.Minute, func() (bool, error) {
+		klog.Infof("Listing pods...")
+		podItems, err := kubeClient.CoreV1().Pods("run-once-duration-override-operator").List(ctx, metav1.ListOptions{})
+		if err != nil {
+			klog.Errorf("Unable to list pods: %v", err)
+			return false, nil
+		}
+		for _, pod := range podItems.Items {
+			if !strings.HasPrefix(pod.Name, "run-once-duration-override-") {
+				continue
+			}
+			klog.Infof("Checking pod: %v, phase: %v, deletionTS: %v\n", pod.Name, pod.Status.Phase, pod.GetDeletionTimestamp())
+			if pod.Status.Phase == corev1.PodRunning && pod.GetDeletionTimestamp() == nil {
+				runOnceDurationoverridePod = pod.DeepCopy()
+				return true, nil
+			}
+		}
+		return false, nil
+	}); err != nil {
+		klog.Errorf("Unable to wait for the RODOO pod to run")
+		os.Exit(1)
+	}
+
+	klog.Infof("Run Once Duration Override running in %v", runOnceDurationoverridePod.Name)
```

Comment:
Also worth checking whether the admission webhook is running as well. I.e. copy paste of https://github.com/openshift/run-once-duration-override/blob/main/test/e2e/operator_test.go#L179-L231